packages/aggregator/src/analytics/aggregateAnalytics/aggregations/getFinalValuePerPeriod.js

```diff
@@ -106,28 +107,60 @@ class FinalValueAggregator {
     this.cache = cache;
   }
 
-  getContinuousValues(analytics, aggregationPeriod) {
-    const periods = getContinuousPeriodsForAnalytics(analytics, aggregationPeriod, true);
+  getContinuousValues(organisationUnitCache, periods) {
+    let mostRecentValue;
 
+    return periods.map(period => {
+      mostRecentValue = organisationUnitCache[period] || mostRecentValue;
+      return {
+        ...mostRecentValue,
+        period,
+      };
+    });
+  }
+
+  fillWithPresetValues(organisationUnitCache, periods, value) {
     const values = [];
-    this.cache.iterateOrganisationUnitCache(organisationUnitCache => {
-      let mostRecentValue;
-
-      periods.forEach(period => {
-        const valueForPeriod = organisationUnitCache[period];
-
-        if (valueForPeriod !== undefined) {
-          mostRecentValue = valueForPeriod;
-        }
-        if (mostRecentValue !== undefined) {
-          values.push({
-            ...mostRecentValue,
-            period,
+    const dataElements = uniq(Object.values(organisationUnitCache).map(data => data.dataElement));
+    const organisationUnits = uniq(
+      Object.values(organisationUnitCache).map(data => data.organisationUnit),
+    );
+
+    periods.forEach(period => {
+      const valueForPeriodFromCache = organisationUnitCache[period];
+      if (valueForPeriodFromCache) {
+        values.push(valueForPeriodFromCache);
+      } else {
+        // Assign empty period with value for each dataElement mapping each orgUnit that existed in results.
+        dataElements.forEach(dataElement => {
+          organisationUnits.forEach(organisationUnit => {
+            values.push({
+              dataElement,
+              organisationUnit,
+              value,
+              period,
+            });
           });
-        }
-      });
+        });
+      }
     });
+    return values;
+  }
+
+  getFilledValues(analytics, aggregationPeriod, fillEmptyPeriodsWith) {
+    const checkIfFillingContinuousValues = fillEmptyPeriodsWith === 'previous';
+    const periods = getContinuousPeriodsForAnalytics(
+      analytics,
+      aggregationPeriod,
+      checkIfFillingContinuousValues,
```

Comment:
I'm not entirely sure if `checkIfFillingContinuousValues` is what we want to be using to determine whether we get periods up until the current period or not... It seems to be that this ought to be another config option though I also feel it's hard to make the argument that we should do this...

Perhaps let's just leave a comment here clarifying why this decision was made?

Reply:
I agree that makes sense to be another config option, but with that we will have more configs in db:
> aggregations: [
      {
        type: 'FINAL_EACH_WEEK_FILL_EMPTY_WEEKS',
        config: {
          fillEmptyPeriodsWith: null,
          continueTilCurrentPeriod: false,
        },
      }

So that is a trade off. TBH I prefer to have more configs because that makes code easier to read. What do you think?

Reply:
Hmm yeah I'm inclined to agree, perhaps something like:
```
{
   fillEmptyPeriodsWith: <any value> (default: 'previous')
   fillEmptyPeriodsTilNow: <boolean> (default: false)
}
```
I'm inclined to make the `fillEmptyPeriodsTilNow` be false by default as imo it does feel a little un-exepected... 

However, that would mean changing the config of all other existing dashboards/overlays that use this aggregation as I think the previous behaviour was to `fillEmptyPeriodsTilNow`... If you're happy to do this then I guess this would be the most complete solution.

Ultimately though, I'm also okay with us using the code you have currently (if fillling `previous` data, then fill til now, otherwise don't) as long as we add a comment explaining that this could/should be expanded to a config in the future? That way the next time that a dev needs to implement divergent logic (either a `previous` fill where they **don't** want to fill til now, or a non-`previous` fill where they **do** want to fill til now) they are suggested to do this 😄

Reply:
Yeah I agree making it be false as default would make more sense. 
I dig in a bit and found this config is used in losts of visuals already. It is doable but I would suggest we can do it in a separate card or leave it for a dev next time.